evaluate/report/csv.go

```diff
@@ -124,3 +82,228 @@ func (e *EvaluationFile) WriteEvaluationRecord(model model.Model, language langu
 func evaluationHeader() (header []string) {
 	return append([]string{"model-id", "model-name", "cost", "language", "repository", "task", "score"}, metrics.AllAssessmentKeysStrings...)
 }
+
+// EvaluationRecord holds a line of the evaluation CSV.
+type EvaluationRecord struct {
+	// ModelID holds the model id.
+	ModelID string
+	// ModelName holds the model name.
+	ModelName string
+	// ModelCost holds the model cost.
+	ModelCost float64
+
+	// LanguageID holds the language id.
+	LanguageID string
+
+	// Assessments holds the assessments of an entry.
+	Assessments metrics.Assessments
+}
+
+// Clone clones an evaluation record.
+func (e *EvaluationRecord) Clone() (new *EvaluationRecord) {
+	new = &EvaluationRecord{}
+
+	new.ModelID = e.ModelID
+	new.ModelName = e.ModelName
+	new.ModelCost = e.ModelCost
+	new.LanguageID = e.LanguageID
+	new.Assessments = metrics.Merge(e.Assessments, nil)
+
+	return new
+}
+
+// EvaluationRecords holds all the evaluation records.
+type EvaluationRecords []*EvaluationRecord
+
+// EvaluationRecordsPerModel holds the collection of evaluation records per model.
+type EvaluationRecordsPerModel map[string]*EvaluationRecord
```

Comment:
Once we get the model costs and names on the fly, this could even be just `map[string][]*metrics.Assessments`.

Reply:
Similar for "by language" below.

Reply:
Follow up